### ISSUE TITLE
inconsistent-return-statements on loop without break statement

### ISSUE THREAD

**Issue by jabdoa2** [PyCQA/pylint#1772]
On pylint 1.8.1, astroid 1.6.0 with this file:

```
"""Test."""
def test():
    """Test function."""
    counter = 1
    while True:
        counter += 1
        if counter == 100:
            return 7

print(test())
```

```
pylint loop.py 
No config file found, using default configuration
************* Module loop
R:  2, 0: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)

------------------------------------------------------------------
Your code has been rated at 8.57/10 (previous run: 8.57/10, +0.00)
```

No other return exists and the loop cannot exit

**Comment by hippo91** [comment]
Thanks for the issue @jabdoa2 .
It is actually a false positive. However fixing it may be hard to achieve. I think it needs to analyze the control flow to be able to detect that the loop will effectively exit when counter equal 100.
@rogalski  @PCManticore what do you think about it?

**Comment by jabdoa2** [comment]
We had this in numerous places. Common in wait_for_xyz asyncio coroutines. In our cases it would be enough to apply this heuristics:

1. while true loop
2. No break statement

**Comment by PCManticore** [comment]
@hippo91 we could exempt while loops since we don't know if they will be completed (and we can try not to get fancy with control flow for this, just don't emit the message if the fault happens in a `while` loop)
